src/components/ha-data-table.ts

```diff
@@ -270,7 +275,11 @@ export class HaDataTable extends BaseElement {
               ),
               (row: DataTabelRowData) => row[this.id],
               (row: DataTabelRowData) => html`
-                <tr data-row-id="${row[this.id]}" class="mdc-data-table__row">
+                <tr
+                  data-row-id="${row[this.id]}"
+                  @click=${this._handleRowClick}
```

Comment:
I think that this is too confusing, you miss tapping the checkbox and get a more info dialog. Can we just attach it to the name or maybe add a button at the end ? 

CC @SeanPM5 for design inspiration.

Reply:
I think it's fine tbh? There seems to be enough room between the checkbox that a miss-tap will be rare, and if you do get the more-info dialog it's easy enough to dismiss. I think that I would prefer this approach rather than having a seperate more info button.

Reply:
I already changed it to just the first column and added a pointer cursor, I think the space between the checkbox and the entity column is quite wide, and adding a button take to much space, especially on mobile.